Makefile

```diff
@@ -274,12 +277,19 @@ showVariables:
 all: showVariables lint debug release apache testdebug testrelease fixrights
 
 .PHONY: user
-user:
+user: env
 	source $(USER_SOURCE) && make all
 
 .PHONY: build
 build: showVariables .build-artefacts/devlibs .build-artefacts/requirements.timestamp $(SRC_JS_FILES) debug release
 
+
+.PHONY: env
+env:
+	curl -o- https://raw.githubusercontent.com/creationix/nvm/$(NVM_VERSION)/install.sh | bash ;\
+	source $(HOME)/.bashrc ;\
+	nvm install $(NODE_VERSION)
```

Comment:
I would try to launch the install command only when you want to update NVM. Otherwise you will always curl nvm program. Create a new target use the `lastvalue` function to store node and nvm versions.

Reply:
Hi @loicgasser , thanks for the feedback. Something like this?

Reply:
After `nvm install`, you have to activate a version `nvm use $(NODE_VERSION)`

Reply:
voilà

Reply:
`.build-artefacts/node-version` will be executed only if `.build-artefacts/last-node-version` has changed so I don't think the if statement is needed.

Reply:
see `cachelastvariable` https://github.com/geoadmin/mf-geoadmin3/blob/master/Makefile#L703

Reply:
didn't work without the `ifndef`, but I might have done something strange

Reply:
Remove the phony from the materialized target

Reply:
If I try like this: 
```
.build-artefacts/nvm-version: .build-artefacts/last-nvm-version
        curl -o- https://raw.githubusercontent.com/creationix/nvm/$(NVM_VERSION)/install.sh | bash

.build-artefacts/node-version: .build-artefacts/last-node-version
        source $(HOME)/.bashrc ;\
        nvm install $(NODE_VERSION)

.PHONY: env
env: .build-artefacts/nvm-version .build-artefacts/node-version
        source $(HOME)/.bashrc && nvm use $(NODE_VERSION)
```
the targets `build-artefacts/nvm-version` and `.build-artefacts/node-version` are always executed

Reply:
yes because
`.build-artefacts/nvm-version` and  `.build-artefacts/node-version` are not materialized actually.
Sticky situation. leave it like this if it works.

Reply:
"ça marche, on touche pas" ...